### PR TITLE
Revert "DD Store Lock"

### PR DESCRIPTION
**Pull request by fabiocos**
Reverts cms-sw/cmssw#24378

The original PR causes a set of failures in IB tests (11624.0, 11626.0, 11642.0, 12024.0, 24034.0).
Temporarily reverting the change, pending a fixed version.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test workflow 11626.0,12024.0,24034.0

File: DetectorDescription/Core/src/DDCompactView.cc [hunk 1]
```diff
@@ -35,19 +35,14 @@ class DDDivision;
...
+    worldpos_( new DDPosData( DDTranslation(), DDRotation(), 0 ))
 {
-  DDMaterial::StoreT::instance().setReadOnly( false );
-  DDSolid::StoreT::instance().setReadOnly( false );
-  DDLogicalPart::StoreT::instance().setReadOnly( false );
-  DDSpecifics::StoreT::instance().setReadOnly( false );
-  DDRotation::StoreT::instance().setReadOnly( false );
-  rep_ = std::make_unique<DDCompactViewImpl>( DDLogicalPart( name ));
-  worldpos_ = std::make_unique<DDPosData>( DDTranslation(), DDRotation(), 0 );
+  DDMaterial::StoreT::instance().setReadOnly(false);
```
Comment (ianna) [inline DetectorDescription/Core/src/DDCompactView.cc]: @fabiocos - this is the only change needed to revert the store lock. I'll submit a new PR for that.

**Comment by ianna** [comment]
@fabiocos - It's probably better if I submit a smaller fix for it - just not to lock the stores:
https://github.com/cms-sw/cmssw/pull/24440#pullrequestreview-151727233

File: DetectorDescription/Core/src/DDCompactView.cc (hunk 1 above)
Comment (fabiocos) [reply DetectorDescription/Core/src/DDCompactView.cc]: @ianna, ok, please do it, so as we may test and integrate it asap

**Comment by ianna** [comment]
@fabiocos - please take this one instead:
https://github.com/cms-sw/cmssw/pull/24441

**Pull request closed without merging**